carta/session.py

```diff
@@ -418,6 +418,57 @@ def clear_raster_scaling_reference(self):
         """Clear the raster scaling reference."""
         self.call_action("clearRasterScalingReference")
 
+    # VIEWER MODES
+    @validate(String())
+    def set_viewer_mode(self, panel="single"):
+        """
+        Switch between single-panel mode and multiple-panel mode.
+
+        Parameters
+        ----------
+        panel : {0}
+            To adopt single-panel mode (``single``) or multiple-panel mode (``multiple``). Default is ``single``.
+        """
+        if panel is "single":
+            if_multiple = False
+        elif panel is "multiple":
+            if_multiple = True
+        else:
+            logger.warning('Cannot proceed with the input panel argument. Please set "single" or "multiple" for the argument.')
+            return
+        self.call_action("widgetsStore.setImageMultiPanelEnabled", if_multiple)
+
+    def previous_page(self):
+        """Go to previous page in viewer."""
+        self.call_action("widgetsStore.onPreviousPageClick")
+
+    def next_page(self):
+        """Go to next page in viewer."""
+        self.call_action("widgetsStore.onNextPageClick")
+
+    @validate(OneOf(*range(1, 101)), OneOf(*range(1, 101)), String())
+    def set_viewer_grid(self, columns=2, rows=2, mode="dynamic"):
```

Comment:
better swap the order of columns and rows to fit the usual convention. ie rows first, then columns

Reply:
this is a set function so it is better not to have default values for the input arguments. Also, columns, rows, and mode are implicit preferences from the frontend point of view.

Reply:
I _think_ when a user call this function to define a grid layout, the mode needs to be "fixed" by default. This would be more consistent with matplotlib.pyplot.subplot(), for example.

Reply:
In e62f8c4:
1. The order between `rows` and `columns` is swapped.
2. Remove the default settings and the descriptions for all arguments _except_ for `mode` (set to `fixed`).

Nevertheless, just a quick question: as for the default value, shouldn't we do what we did for the "set contour" or "set vector overlay" issue, that we can execute the function without any parameter set (like: `session.set_viewer_grid()`) then get the result of the default setting of the frontend (2x2 grids, dynamic mode)?